daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -803,6 +803,54 @@ case class FNFloor(recipe: CompiledDPath, argType: NodeInfo.Kind) extends FNOneA
   }
 }
 
+case class DFDLXLeftShift(recipes: List[CompiledDPath],argType: NodeInfo.Kind) extends FNTwoArgs(recipes) {
+  override def computeValue(arg1: DataValuePrimitive, arg2: DataValuePrimitive, dstate: DState) = {
+    val shift = asInt(arg2.getAnyRef).intValue
+    argType match {
+      case NodeInfo.Long => {
+        val leftShift = asLong(arg1).longValue()
+        if (leftShift.toString.length < shift) {
```

Comment:
You don't want the toString length here. You want 64 because a Long is defined to be 64 bits long. 
Similarly below for the int case you want 32, short 16, byte 8. 

I think the comparison wants to be <=, because if the shift left is 64 or greater, then the answer is 0.

Reply:
I have changed it to if shift>=64 then ans is 0.